assume/world.py

```diff
@@ -213,7 +216,12 @@ def add_unit_operator(
         unit_operator_agent._role_context.data_dict = {
             "output_agent_addr": self.output_agent_addr[0],
             "output_agent_id": self.output_agent_addr[1],
-            "learning_mode": self.learning_mode,
+            "learning_agent_addr": self.learning_agent_addr[0]
+            if self.learning_mode
+            else None,
+            "learning_agent_id": self.learning_agent_addr[1]
+            if self.learning_mode
+            else None,
```

Comment:
```suggestion
        unit_operator_agent._role_context.data_dict = {}
        if not self.learning_mode:
            unit_operator_agent._role_context.data_dict.update({
                "output_agent_addr": self.output_agent_addr[0]
                "output_agent_id": self.output_agent_addr[1]
            })
        else:
            unit_operator_agent._role_context.data_dict.update({
                "learning_agent_addr": self.learning_agent_addr[0]
                "learning_agent_id": self.learning_agent_addr[1]
            })
```

something like this..?